src/kernels/kernel.ts

```diff
@@ -151,7 +152,8 @@ export class Kernel implements IKernel {
     private startCancellation = new CancellationTokenSource();
     private startupUI = new DisplayOptions(true);
     constructor(
-        public readonly notebookDocument: NotebookDocument,
+        public readonly notebookDocument: NotebookDocument | undefined,
```

Comment:
I was hoping we could make this private? The fact that they're tied together seems like it might be private.

Reply:
I guess I could, but then I'd have to modify all of the followig code:
```nb = kernel.notebookdocument```
with
```nb = workspace.notebooks.find(nb => nb.uri.toString() === kernel.owner.uri.toString)```